### PR TITLE
Updated free surface displacements for GPUs

### PR DESCRIPTION
**Pull request by ravil-mobile**

### CODE REVIEW

File: src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp [hunk 1]
```diff
@@ -145,24 +145,36 @@ void LocalIntegrationRecorder::recordLocalFluxIntegral() {
...
 void LocalIntegrationRecorder::recordDisplacements() {
-  real **displacements = currentLayer->var(currentHandler->displacements);
-  std::vector<real *> iVelocitiesPtrs{};
-  std::vector<real *> displacementsPtrs{};
+  real *(*faceDisplacements)[4] = currentLayer->var(currentHandler->faceDisplacements);
+  std::array<std::vector<real *>, 4> iVelocitiesPtrs{{}};
+  std::array<std::vector<real *>, 4> displacementsPtrs{};
+
+  constexpr unsigned numStressComponents{6};
+  constexpr unsigned offsetToVelocities = numStressComponents * (init::I::Stop[0] - init::I::Start[0]);
```
Comment (ravil-mobile) [inline src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp:153]: There should be a better way to compute the tensor stride of the 1st dimension
```
init::I::Stop[0] - init::I::Start[0]
``` 
Please, suggest something

File: src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp [hunk 2]
```diff
@@ -145,24 +145,36 @@ void LocalIntegrationRecorder::recordLocalFluxIntegral() {
...
-      real *iVelocity = &idofsAddressRegistry[cell][OFFSET_TO_VELOCITIES];
-      iVelocitiesPtrs.push_back(iVelocity);
-      displacementsPtrs.push_back(displacements[cell]);
+    auto data = currentLoader->entry(cell);
+
+    for (unsigned face = 0; face < 4; ++face) {
+      auto isRequired = faceDisplacements[cell][face] != nullptr;
+      auto notFreeSurfaceGravity = data.cellInformation.faceTypes[face] != FaceType::freeSurfaceGravity;
+
+      if (isRequired && notFreeSurfaceGravity) {
```
Comment (ravil-mobile) [inline src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp:164]: @krenzland , could you, please, whether conditions are ok?

Comment (krenzland) [reply src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp:164]: Yup, they are correct.

File: src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp (hunk 1 above)
Comment (krenzland) [reply src/Initializer/BatchRecorders/LocalIntegrationRecorder.cpp:153]: Hm, I think this is fine. Don't know a better way right now

**Pull request merged**